docs/cryptodoc/src/05_10_cmce.rst

```diff
@@ -0,0 +1,485 @@
+<pubkey/cmce/component_table>` shows an overview of the components and
+their corresponding file locations. Classic McEliece, like all Key Encapsulation
+Mechanisms (KEMs), consists of three main algorithms: Key generation, key
+encapsulation, and key decapsulation. In the case of Classic McEliece, the key
+generation process involves creating a field ordering and an irreducible
+polynomial. These two components define the randomly chosen Goppa code, the
+secret information stored in the private key. A parity check matrix in
+systematic form is derived from the Goppa code to generate the public key used
+for encapsulation. More detailed information about these processes can be found
+in the following sections.
+
+.. _pubkey/cmce/component_table:
+
+.. table:: Classic McEliece components and file locations. Its files are located at :srcref:`src/lib/pubkey/classic_mceliece/`.
+   :widths: 21 29 40
+
+   +--------------------------------------------------------+------------------------------------------------------------------+----------------------------------------------------------------------------+
+   | Component                                              | File                                                             | Purpose                                                                    |
+   +========================================================+==================================================================+============================================================================+
+   | :ref:`Types <pubkey/cmce/types>`                       | :srcref:`[src/lib/pubkey/classic_mceliece]/cmce_types.h`         | Strong types                                                               |
+   +--------------------------------------------------------+------------------------------------------------------------------+----------------------------------------------------------------------------+
+   | :ref:`Galois Field Arithmetic <pubkey/cmce/gf>`        | :srcref:`[src/lib/pubkey/classic_mceliece]/cmce_gf.h`            | Galois field data structure and logic                                      |
+   +--------------------------------------------------------+------------------------------------------------------------------+----------------------------------------------------------------------------+
+   | :ref:`Field Ordering <pubkey/cmce/field_ordering>`     | :srcref:`[src/lib/pubkey/classic_mceliece]/cmce_field_ordering.h`| Field ordering creation and Beneš network computation                      |
+   +--------------------------------------------------------+------------------------------------------------------------------+----------------------------------------------------------------------------+
+   | :ref:`Polynomial Arithmetic <pubkey/cmce/poly>`        | :srcref:`[src/lib/pubkey/classic_mceliece]/cmce_poly.h`          | Polynomial data structure, logic, and minimal polynomial creation          |
```

Comment:
Die Links zeigen hier scheinbar auf den Rohde-Schwarz fork von Botan und ich bekomme dort die Fehlermeldung: "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."